File: src/tui.rs
use anyhow::Result;
use console::Style;
use dialoguer::{Confirm, Input, Password, Select};

const LOGO: &str = r#"              __  __                     _     
  ___  ____  / /_/ /_  _________  ____  (_)____
 / _ \/ __ \/ __/ __ \/ ___/ __ \/ __ \/ / ___/
/  __/ / / / /_/ / / / /  / /_/ / /_/ / / /__  
\___/_/ /_/\__/_/ /_/_/   \____/ .___/_/\___/  
                              /_/              "#;

const SEPARATOR: &str = "──────────────────────────────────────────────────────────────";

pub fn pink() -> Style {
    // 219 = #ffafff — soft pink/rose, 211 = #ff87af — deeper pink
    Style::new().color256(219)
}

pub fn dimmed() -> Style {
    Style::new().dim()
}

pub fn bold_white() -> Style {
    Style::new().bold()
}

pub fn success_green() -> Style {
    Style::new().green()
}

pub fn error_red() -> Style {
    Style::new().red()
}

pub fn print_header() {
    let p = pink();
    for line in LOGO.lines() {
        println!("{}", p.apply_to(line));
    }
    println!(
        "🧠  {}          {}",
        dimmed().apply_to("spec-driven development"),
        dimmed().apply_to("v0.1.0")
    );
    println!("{}", p.apply_to(SEPARATOR));
    println!();
}

pub fn print_success(msg: &str) {
    println!("{} {}", success_green().apply_to("✓"), msg);
}

pub fn print_error(msg: &str) {
    eprintln!("{} {}", error_red().apply_to("✗"), msg);
}

pub fn print_info(msg: &str) {
    println!("{}", bold_white().apply_to(msg));
}

pub fn print_dim(msg: &str) {
    println!("{}", dimmed().apply_to(msg));
}

pub fn confirm(prompt: &str) -> Result<bool> {
    let result = Confirm::new()
        .with_prompt(prompt)
        .default(false)
        .interact()?;
    Ok(result)
}

pub fn input(prompt: &str) -> Result<String> {
    let result: String = Input::new()
        .with_prompt(prompt)
        .interact_text()?;
    Ok(result)
}

pub fn input_with_default(prompt: &str, default: &str) -> Result<String> {
    let result: String = Input::new()
        .with_prompt(prompt)
        .default(default.to_string())
        .interact_text()?;
    Ok(result)
}

pub fn password(prompt: &str) -> Result<String> {
    let result = Password::new()
        .with_prompt(prompt)
        .interact()?;
    Ok(result)
}

pub fn select(prompt: &str, items: &[&str]) -> Result<usize> {
    let result = Select::new()
        .with_prompt(prompt)
        .items(items)
        .default(0)
        .interact()?;
    Ok(result)
}

pub fn select_string(prompt: &str, items: &[String]) -> Result<usize> {
    let refs: Vec<&str> = items.iter().map(|s| s.as_str()).collect();
    select(prompt, &refs)
}

File: src/tui.ts
import chalk from 'chalk';
import { confirm as inquirerConfirm, input as inquirerInput, password as inquirerPassword, select as inquirerSelect } from '@inquirer/prompts';

const LOGO = `   _____            ____  
  / ___/____  ___  / __ \\
  \\__ \\/ __ \\/ _ \\/ / / /
 ___/ / /_/ /  __/ /_/ / 
/____/ .___/\\___/\\___\\_\\ 
    /_/                  `;

const SEPARATOR = '──────────────────────────────────────────────────────────────';

export function pink(text: string): string {
  return chalk.ansi256(219)(text);
}

export function dimmed(text: string): string {
  return chalk.dim(text);
}

export function boldWhite(text: string): string {
  return chalk.bold.white(text);
}

export function successGreen(text: string): string {
  return chalk.green(text);
}

export function errorRed(text: string): string {
  return chalk.red(text);
}

export function warnYellow(text: string): string {
  return chalk.yellow(text);
}

export function boldYellow(text: string): string {
  return chalk.bold.yellow(text);
}

export function boldText(text: string): string {
  return chalk.bold(text);
}

export function removed(text: string): string {
  return chalk.red(text);
}

export function added(text: string): string {
  return chalk.green(text);
}

export function printHeader(): void {
  console.log();
  console.log(pink(LOGO));
  console.log();
  console.log(pink('  🧠  true spec-driven development') + chalk.dim('          v0.1.0'));
  console.log(pink(SEPARATOR));
  console.log();
}

export function printSuccess(msg: string): void {
  console.log(`${chalk.green('✓')} ${msg}`);
}

export function printError(msg: string): void {
  console.error(`${chalk.red('✗')} ${msg}`);
}

export function printDim(msg: string): void {
  console.log(chalk.dim(msg));
}

export async function confirm(prompt: string): Promise<boolean> {
  return inquirerConfirm({ message: prompt });
}

export async function input(prompt: string): Promise<string> {
  return inquirerInput({ message: prompt });
}

export async function inputWithDefault(prompt: string, defaultValue: string): Promise<string> {
  return inquirerInput({ message: prompt, default: defaultValue });
}

export async function password(prompt: string): Promise<string> {
  return inquirerPassword({ message: prompt, mask: '*' });
}

export function printWorkdir(dir: string): void {
  console.log(chalk.dim('  📁  ') + chalk.ansi256(219)(dir));
  console.log();
}

export async function pressEnter(): Promise<void> {
  await inquirerInput({ message: chalk.dim('Press Enter to continue…') });
}

export async function select(prompt: string, items: string[]): Promise<number> {
  const choices = items.map((name, value) => ({ name, value }));
  return inquirerSelect({ message: prompt, choices });
}
